### PR TITLE
add cookbook for mkl regression

### PR DESCRIPTION
**Pull request by sanuj**
Will send integration data once code is reviewed.

### CODE REVIEW

File: doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
+=======================================
+Multiple Kernel Learning for Regression
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst]: remove: regression

File: doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+=======================================
+
+Multiple kernel learning (MKL) is based on convex combinations of arbitrary kernels over potentially different domains.
+
+.. math::
+
+    {\bf k}(x_i,x_j)=\sum_{i=1}^{K} \beta_k {\bf k}_i(x_i, x_j)
+
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst]: do we have a SVM regression cookbook that we can reference here?

File: examples/meta/src/regression/multiple_kernel_learning_regression.sg [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#![create_features]
+RealFeatures features_train(f_feats_train)
+RealFeatures features_test(f_feats_test)
+RegressionLabels labels_train(f_labels_train)
+RegressionLabels labels_test(f_labels_test)
+#![create_features]
+
+#![create_kernel]
+PolyKernel poly_kernel(10,2)
```
Comment (karlnapf) [inline examples/meta/src/regression/multiple_kernel_learning_regression.sg]: no precomputed kernel here? :)

File: doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+where :math:`\beta_k > 0`, :math:`\sum_{k=1}^{K} \beta_k = 1`, :math:`K` is the number of sub-kernels, :math:`\bf{k}` is a combined kernel, :math:`{\bf k}_i` is an individual kernel and :math:`{x_i}_i` are the training data.
+
+Regression is done by using :sgclass:`CSVMLight`.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst]: MKL regression needs an SVM solver as input. We here use SVMLight. 


Put this further down

File: doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Then we create indvidual kernels like :sgclass:`CPolyKernel` and :sgclass:`CGaussianKernel` which will be later combined in one :sgclass:`CCombinedKernel`.
+
+.. sgexample:: multiple_kernel_learning_regression.sg:create_kernel
+
+We create an instance of :sgclass:`CCombinedKernel` and append the :sgclass:`CKernel` objects.
+
+.. sgexample:: multiple_kernel_learning_regression.sg:create_combined_train
+
+We create an object of :sgclass:`SVMLight` and :sgclass:`CMKLRegression`, provide the combined kernel and labels before training it.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst]: Put the SVMLight as SVM solver sentence hre

File: examples/meta/src/regression/multiple_kernel_learning_regression.sg [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+#![create_combined_train]
+CombinedKernel combined_kernel()
+combined_kernel.append_kernel(poly_kernel)
+combined_kernel.append_kernel(gauss_kernel_1)
+combined_kernel.append_kernel(gauss_kernel_2)
+combined_kernel.init(features_train, features_train)
+#![create_combined_train]
+
+#![train_mkl]
+SVRLight svr_constraints()
```
Comment (karlnapf) [inline examples/meta/src/regression/multiple_kernel_learning_regression.sg]: call this binary_svm_solver

**Comment by sanuj** [comment]
@karlnapf I have updated this PR. If you are fine with the code then merge the data PR.

**Comment by karlnapf** [comment]
Data merged, you can update

File: doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst [hunk 7]
```diff
@@ -0,0 +1,56 @@
+========================
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/multiple_kernel_learning_regression.rst]: can you remove "regression" from filename
Ah crap, need to change data filename as well, didnt realise earlier

File: examples/meta/src/regression/multiple_kernel_learning_regression.sg [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+RealVector alpha = mkl.get_alphas()
+#![extract_weights]
+
+#![mkl_apply]
+combined_kernel.init(features_train, features_test)
+RegressionLabels labels_predict = mkl.apply_regression()
+#![mkl_apply]
+
+#![evaluate_error]
+MeanSquaredError eval()
```
Comment (karlnapf) [inline examples/meta/src/regression/multiple_kernel_learning_regression.sg]: pls dont call this eval

**Comment by karlnapf** [comment]
Ok, need change the filename and some mini updates, then update data, then good to merge

**Comment by sanuj** [comment]
@karlnapf good to be merged.

**Pull request merged**